multisig-wallet/project/multisig-contract/src/main.sw

```diff
@@ -98,6 +102,33 @@ impl MultiSignatureWallet for Contract {
         });
     }
 
+    #[storage(read, write)]
+    fn set_threshold(
+        data: b256,
+        nonce: u64,
+        signatures: Vec<SignatureInfo>,
+        threshold: u64,
+    ) {
+        require(storage.nonce != 0, InitError::NotInitialized);
+        require(threshold != 0, InitError::ThresholdCannotBeZero);
+        require(threshold <= storage.total_weight, InitError::TotalWeightCannotBeLessThanThreshold);
+
+        let transaction_hash = create_hash(data, nonce, Identity::ContractId(contract_id()), 0);
```

Comment:
Yeah, I guess that's an oversight on my part. I'll update to remove the nonce as an argument from this function. This way they need to provide the correct nonce in the initial tx creation but not here so there is a time restriction as the hash will not match if another tx is executed first.

I'll do that as a separate issue so that this can be merged now.